Assets/Scripts/Layers/Adapters/DataTypeAdapters/CSVImportAdapter.cs

```diff
@@ -14,14 +14,37 @@
 namespace Netherlands3D.Twin
 {
     [CreateAssetMenu(menuName = "Netherlands3D/Adapters/CSVImportAdapter", fileName = "CSVImportAdapter", order = 0)]
-    public class CSVImportAdapter : ScriptableObject
+    public class CSVImportAdapter : ScriptableObject, IDataTypeAdapter
     {
         [SerializeField] private UnityEvent<string> csvReplacedMessageEvent = new();
         [SerializeField] private UnityEvent<float> progressEvent = new();
         public int maxParsesPerFrame = 100;
         private static DatasetLayer activeDatasetLayer; //todo: allow multiple datasets to exist
 
-        public void ParseCSVFile(string file)
+        readonly CsvConfiguration config = new(CultureInfo.CurrentCulture)
+        {
+            HasHeaderRecord = true,
+            Delimiter = ";"
+        };
+
+        public bool Supports(LocalFile localFile)
+        {
+            if(!localFile.LocalFilePath.ToLower().EndsWith(".csv"))
+                return false;
+
+            // Check if we can read the CVS using our expected config
+            using var reader = new StreamReader(localFile.LocalFilePath);
```

Comment:
Wouldn't this mean we will parse the file twice? Wouldn't it be sufficient to do this in the Execute and use an exception? This pattern may also allow for the supports method to continue to be a boolean and use exceptions in the Execute method to say "This is the correct adapter, but it is invalid"?

Reply:
Also: this will read the file in a blocking manner, causing performance issues when using large files

Reply:
Yes good idea, didnt think about big .csv files making the check.

Reply:
Renaming import to be specific to building colors, and add header check in supports method.